### ISSUE TITLE
Unable to run some native Linux games

### ISSUE THREAD

**Issue by Kabouik** [tkashkin/GameHub#267]
Some of my Linux games imported into Gamehub fail to run, although executing the binary file directly from Nautilus or Terminal works correctly. When running them from Gamehub, they appear as "running" for a second or so in the interface, but no window shows up and the "running" status goes away. I am not sure how to troubleshoot and get logs to see what are the errors because the terminal command displayed in the game details does not work on my system (perhaps because I'm using the AppImage and `com.github.tkashkin.gamehub` is not recognized?).

###### Steps to reproduce
1. Import a native Linux game, point to the executable file
2. Try to run the game with Gamehub, see that if fails with some files (not all, but I don't know what is the causal factor yet)
3. Try to run the game without Gamehub, it works

Note that I think it can also fail for a few GOG games installed using Gamehub: I seem to remember [GOG] Prison Architect failed to run as well despite clean installation using Gamehub. However, I cannot check before going home tonight. I will edit accordingly.

###### Version and environment

```
Version: 0.14.0-371a9c9-dev
Branch:  dev
Commit:  371a9c9 (371a9c9ed92a898f4e3ef0b8b2958f509fb0a01d)
Distro:  [AppImage] Solus
DE:      Budgie:GNOME
```

###### GDB log

`com.github.tkashkin.gamehub --gdb` does not work on my system, perhaps because I am using the AppImage. I tried `/path/to/AppImagefile --gdb` as well with no success. I guess everything will be easier to solve once I know how to run games from terminal using the AppImage version of Gamehub.

**Comment by tkashkin** [comment]
Run GameHub in terminal with `-d --verbose` parameters.

**Comment by Kabouik** [comment]
Unfortunately that doesn't seem to work with the AppImage:

```bash
kabouik@solus-desktop ~/.config/gamehub $ ./GameHub-bionic-0.14.0-39-dev-371a9c9-x86_64.AppImage -d --verbose
[AppRun ] GameHub AppImage
[AppRun ] AppDir: /tmp/.mount_GameHuoOWGXc
[CheckRT] Checking library versions
[CheckRT] libstdc++.so.6: sys: GLIBCXX_3.4.25; app: GLIBCXX_3.4.26
[CheckRT] libgcc_s.so.1: sys: GCC_7.0.0; app: GCC_7.0.0
[CheckRT] LD_LIBRARY_PATH: /tmp/.mount_GameHuoOWGXc/usr/optlib/libstdc++.so.6::/tmp/.mount_GameHuoOWGXc/usr/lib/x86_64-linux-gnu/webkit2gtk-4.0:/tmp/.mount_GameHuoOWGXc/usr/lib:
[CheckRT] LD_PRELOAD: /tmp/.mount_GameHuoOWGXc/usr/optlib/exec.so:
[AppRun ] Starting GameHub
kabouik@solus-desktop ~/.config/gamehub $ 
```

**Comment by tkashkin** [comment]
It may be related to AppImage. Try to unpack it with `--appimage-extract`. Then run GameHub from extracted directory.
You will need to set a few environment variables to run it (at least `GSETTINGS_SCHEMA_DIR` and probably `LD_LIBRARY_PATH`). See https://github.com/tkashkin/GameHub/blob/dev/scripts/AppRun

It should be something like this:
```bash
./GameHub*.AppImage --appimage-extract
cd squashfs-root
export GSETTINGS_SCHEMA_DIR="`pwd`/usr/share/glib-2.0/schemas/:$GSETTINGS_SCHEMA_DIR"
export LD_LIBRARY_PATH="`pwd`/usr/lib/x86_64-linux-gnu/webkit2gtk-4.0:$LD_LIBRARY_PATH"
cd usr
./bin/com.github.tkashkin.gamehub
```

**Comment by Kabouik** [comment]
I'm still missing something, not sure how to fix it:

```bash
kabouik@solus-desktop ~/.config/gamehub/squashfs-root/usr $ ./bin/com.github.tkashkin.gamehub
[ERROR]  [GLib-GIO] Settings schema 'com.github.tkashkin.gamehub.paths.collection' is not installed
Trace/breakpoint trap
```

**Comment by tkashkin** [comment]
```bash
GSETTINGS_SCHEMA_DIR="`pwd`/share/glib-2.0/schemas/:$GSETTINGS_SCHEMA_DIR" ./bin/com.github.tkashkin.gamehub -d --verbose
```

**Comment by Kabouik** [comment]
I am getting the exact same error with this command line.

**Comment by tkashkin** [comment]
Try to remove `checkrt.sh` and use `AppRun` script. It should setup everything the same way like inside AppImage, but skip overriding some libraries:
```bash
cd squashfs-root
mv checkrt.sh checkrt.sh.skip
APPDIR="`pwd`" ./AppRun -d --verbose
```

**Comment by Kabouik** [comment]
Thanks, I could run GameHub that way indeed. It solved the itch.io client issue (for the record, the window did show up, but was empty), but not the games that just run for a split second:

```bash
kabouik@solus-desktop ~/.config/gamehub/squashfs-root $ APPDIR="`pwd`" ./AppRun -d --verbose
[AppRun ] GameHub AppImage
[AppRun ] AppDir: /home/kabouik/.config/gamehub/squashfs-root
[AppRun] Skipping CheckRT
[AppRun ] Starting GameHub
[DEBUG]  [Database.migrate] Latest db version: 8, current: 8
[DEBUG]  [WineWrap] Updating index
[INFO]   Version: 0.14.0-371a9c9-dev
[INFO]   Branch:  dev
[INFO]   Commit:  371a9c9 (371a9c9ed92a898f4e3ef0b8b2958f509fb0a01d)
[INFO]   Distro:  [AppImage] Solus
[INFO]   DE:      Budgie:GNOME
[DEBUG]  [Gamepad] 'Microsoft X-Box 360 pad' connected
[DEBUG]  [GOG] Loading games: page 1 of 1
[DEBUG]  [GOG] Imported tag: COMPLETED (249723123)
[DEBUG]  [GOG] Imported tag: NEXT TO PLAY (249723113)
[DEBUG]  [GOG] Imported tag: BACKLOG (249723103)
[DEBUG]  [GOG] Imported tag: FAVORITE (249723093)
[FATAL]  [Json] json_object_get_int_member: assertion 'JSON_NODE_TYPE (node) == JSON_NODE_VALUE' failed
[FATAL]  [Json] json_object_get_int_member: assertion 'JSON_NODE_TYPE (node) == JSON_NODE_VALUE' failed
[DEBUG]  [WineWrap] Index updated
[DEBUG]  [Utils.run] {'/run/media/kabouik/INT4TB/JEUX/AUTRES/Gone home/gonehome-linux/GoneHome_linux.x86_64'}; dir: '/run/media/kabouik/INT4TB/JEUX/AUTRES/Gone home/gonehome-linux'
Found path: /run/media/kabouik/INT4TB/JEUX/AUTRES/Gone home/gonehome-linux/GoneHome_linux.x86_64
Mono path[0] = '/run/media/kabouik/INT4TB/JEUX/AUTRES/Gone home/gonehome-linux/GoneHome_linux_Data/Managed'
Mono path[1] = '/run/media/kabouik/INT4TB/JEUX/AUTRES/Gone home/gonehome-linux/GoneHome_linux_Data/Mono'
Mono config path = '/run/media/kabouik/INT4TB/JEUX/AUTRES/Gone home/gonehome-linux/GoneHome_linux_Data/Mono/etc'
displaymanager : xrandr version warning. 1.6
client has 6 screens
displaymanager screen (0)(DVI-D-0): 1920 x 1200
displaymanager screen (1)(HDMI-A-3): 1080 x 1920
Using libudev for joystick management


Importing game controller configs
Found /dev/input/event23
Mapping raw axis 0 to 0
Mapping raw axis 1 to 1
Mapping raw axis 2 to 2
Mapping raw axis 3 to 3
Mapping raw axis 4 to 4
Mapping raw axis 5 to 5
Microsoft X-Box 360 pad: Mapping b0.0 to b0
Microsoft X-Box 360 pad: Mapping b1.0 to b1
Microsoft X-Box 360 pad: Mapping b6.0 to b6
Microsoft X-Box 360 pad: Mapping h0.4 to a7
Microsoft X-Box 360 pad: Mapping h0.8 to a6
Microsoft X-Box 360 pad: Mapping h0.2 to a6
Microsoft X-Box 360 pad: Mapping h0.1 to a7
Microsoft X-Box 360 pad: Mapping b8.0 to b8
Microsoft X-Box 360 pad: Mapping b4.0 to b4
Microsoft X-Box 360 pad: Mapping b9.0 to b9
Microsoft X-Box 360 pad: Mapping a2.0 to a2
Microsoft X-Box 360 pad: Mapping a0.0 to a0
Microsoft X-Box 360 pad: Mapping a1.0 to a1
Microsoft X-Box 360 pad: Mapping b5.0 to b5
Microsoft X-Box 360 pad: Mapping b10.0 to b10
Microsoft X-Box 360 pad: Mapping a5.0 to a5
Microsoft X-Box 360 pad: Mapping a3.0 to a3
Microsoft X-Box 360 pad: Mapping a4.0 to a4
Microsoft X-Box 360 pad: Mapping b7.0 to b7
Microsoft X-Box 360 pad: Mapping b2.0 to b2
Microsoft X-Box 360 pad: Mapping b3.0 to b3
Assigning joystick 1
[FATAL]  [Json] json_object_get_object_member: assertion 'node != NULL' failed
[FATAL]  [Json] json_object_get_string_member: assertion 'object != NULL' failed
[WARN]   [ImageCache] Error loading image 'http://cdn.akamai.steamstatic.com/steam/apps/774941/header.jpg': Not Found
[WARN]   [ImageCache] Error loading image 'http://cdn.akamai.steamstatic.com/steam/apps/205790/header.jpg': Not Found
[WARN]   [ImageCache] Error loading image 'http://media.steampowered.com/steamcommunity/public/images/apps/205790/.jpg': Not Found
[WARN]   [ImageCache] Error loading image 'http://cdn.akamai.steamstatic.com/steam/apps/737230/header.jpg': Not Found
[WARN]   [ImageCache] Error loading image 'http://cdn.akamai.steamstatic.com/steam/apps/700580/header.jpg': Not Found
[WARN]   [ImageCache] Error loading image 'http://cdn.akamai.steamstatic.com/steam/apps/236830/header.jpg': Not Found
[FATAL]  [Json] json_object_get_object_member: assertion 'node != NULL' failed
[FATAL]  [Json] json_object_get_string_member: assertion 'object != NULL' failed
[DEBUG]  [Utils.run] {'/run/media/kabouik/INT4TB/JEUX/AUTRES/Prison Architect/start.sh'}; dir: '/run/media/kabouik/INT4TB/JEUX/AUTRES/Prison Architect'
Running Prison Architect
Suggested default screen resolution of 1920 x 1200 on display 1
WindowManagerSDL2 :: Main
SDL Version: Compiled against 2.0.3, running with 2.0.3
User save path set to '/home/kabouik/.Prison Architect/'
multiplayer-4f linux gog compiled 18:14:17, Dec 18 2018
Loading Preferences from /home/kabouik/.Prison Architect/preferences.txt
Failed to find /home/kabouik/.Prison Architect/preferences.txt
Suggested resolution : 1920 x 1200
Opening window 1920 x 1200 x 32, fullscreen 
Opening on display 0 with bounds (0, 447) - (1920 x 1200)
Screen 1920 x 1200, bpp 24 (High DPI Scale 1.000000, 1.000000)
OpenGL Vendor     : (null)
OpenGL Renderer   : (null)
OpenGL Version    : (null)
OpenGL GLSL       : (null)
./PrisonArchitect: line 59:  8801 Segmentation fault      "$base"/PrisonArchitect.x86_64 "$@"
[DEBUG]  [Utils.run] {'/run/media/kabouik/INT4TB/JEUX/AUTRES/Overgrowth/Overgrowth.bin.x86_64'}; dir: '/run/media/kabouik/INT4TB/JEUX/AUTRES/Overgrowth'
[i][__]: filesystem.cpp: 485: Adding data path /run/media/kabouik/INT4TB/JEUX/AUTRES/Overgrowth/
[i][__]:platformsetup.cpp:  51: Not adding /run/media/kabouik/INT4TB/JEUX/AUTRES/Overgrowth/ to FindPath list as it's the same as ./
[i][__]: filesystem.cpp: 476: Adding write path /home/kabouik/.local/share/Overgrowth/
Preserving 1854986 of data from previous log file, starting at pos 0
[i][__]:       main.cpp:  89: Starting program. Version b5.2-dirty_build-4403 2017-09-04 11:42:43 UTC  amd64 Linux
[i][__]:       main.cpp:  92: Deploy (Release)
Initializing Theora Playback Library (1.0 RC2)
[S_API FAIL] SteamAPI_Init() failed; no appID found.
Either launch the game from Steam, or put the file steam_appid.txt containing the correct appID in your game folder.
[i][__]: steamworks.cpp:  54: Could not connect to Steam API.
[i][__]:     engine.cpp:7567: Initializing SDL
[i][__]:     engine.cpp:7569: SDL Initialized successfully
[i][__]:      input.cpp: 305: Added 1 controller mappings.
[i][__]:      input.cpp: 152: Attached controller 0:X360 Controller
[i][__]:      input.cpp: 255: Attached joystick 0:Microsoft X-Box 360 pad
[i][__]:      input.cpp:  43: Current controllers:
[i][__]:      input.cpp:  47: Joystick0,Microsoft X-Box 360 pad
[i][__]:   graphics.cpp: 726: There are 3 available video drivers.
[i][__]:   graphics.cpp: 727: Video drivers are following:
[i][__]:   graphics.cpp: 731: Video driver 0: x11
[i][__]:   graphics.cpp: 731: Video driver 1: wayland
[i][__]:   graphics.cpp: 731: Video driver 2: dummy
[i][__]:   graphics.cpp: 738: Initialized video driver: x11
[i][__]:   graphics.cpp: 755: Attribute group 1 set
[i][__]:   graphics.cpp: 774: Created window
[i][__]:   graphics.cpp: 776: Set minimum size
[i][__]:   graphics.cpp: 784: Set display mode
[i][__]:   graphics.cpp: 789: Created 3.2 GL Context
[i][__]:   graphics.cpp: 802: RGBA bits, depth: 8 8 8 8 24
[i][__]:   graphics.cpp: 809: Anti-aliasing samples: 0
[f][__]:crashreport.cpp: 241: Game crashed, flushing the log systems. DumpFile: /tmp/38f3e415-821e-970a-62a4a8a1-22ca1a06.dmp
[f][__]:crashreport.cpp: 265: /tmp/38f3e415-821e-970a-62a4a8a1-22ca1a06.dmp /home/kabouik/.local/share/Overgrowth//crash_b5.2-dirty_build-4403_19-06-28_23-35-45/38f3e415-821e-970a-62a4a8a1-22ca1a06.dmp
[i][__]:      error.cpp: 161: Displaying message: Fatal Error, A fatal error occured! Please compress the latest crash folder into a .zip and send to bugs@wolfire.com, please include a description of what exactly was happening in the game when this error occured.
[i][__]:      error.cpp: 162: /run/media/kabouik/INT4TB/JEUX/AUTRES/Overgrowth(_Z12DisplayErrorPKcS0_9ErrorTypeb+0xf8) [0x9e7b08]
/run/media/kabouik/INT4TB/JEUX/AUTRES/Overgrowth() [0x9dc74f]
/run/media/kabouik/INT4TB/JEUX/AUTRES/Overgrowth(_ZN15google_breakpad16ExceptionHandler12GenerateDumpEPNS0_12CrashContextE+0x3d4) [0xf476f4]
/run/media/kabouik/INT4TB/JEUX/AUTRES/Overgrowth(_ZN15google_breakpad16ExceptionHandler13SignalHandlerEiP7siginfoPv+0x97) [0xf47a07]
/usr/lib/libpthread.so.0(+0x12df0) [0x7f976459ddf0]

[e][__]:os_dialogs_linux.cpp:  49: Failed to show message dialog: No message system available
[DEBUG]  [Utils.run] {'/run/media/kabouik/INT4TB/JEUX/AUTRES/Low light combat/llc_linux_64/Low Light Combat (64-bit)/Low Light Combat'}; dir: '/run/media/kabouik/INT4TB/JEUX/AUTRES/Low light combat/llc_linux_64/Low Light Combat (64-bit)'
Found path: /run/media/kabouik/INT4TB/JEUX/AUTRES/Low light combat/llc_linux_64/Low Light Combat (64-bit)/Low Light Combat
Mono path[0] = '/run/media/kabouik/INT4TB/JEUX/AUTRES/Low light combat/llc_linux_64/Low Light Combat (64-bit)/Low Light Combat_Data/Managed'
Mono path[1] = '/run/media/kabouik/INT4TB/JEUX/AUTRES/Low light combat/llc_linux_64/Low Light Combat (64-bit)/Low Light Combat_Data/Mono'
/dev/input/js0: driver version: 2.1.0 (20100)
/dev/input/js0: fd 3, buttons 11, axes 8, name Microsoft X-Box 360 pad
/dev/input/js0: using XBox 360 Controller mappings
/dev/input/js0: axis  0: raw  -7234, mapped 0.000000
/dev/input/js0: axis  1: raw  -1539, mapped 0.000000
/dev/input/js0: axis  3: raw    766, mapped 0.000000
/dev/input/js0: axis  4: raw      0, mapped 0.000000
/dev/input/js0: axis  6: raw      0, mapped 0.000000
/dev/input/js0: axis  7: raw      0, mapped 0.000000
[DEBUG]  [Utils.run] {'/home/kabouik/.itch/app-25.4.0/itch'}; dir: '/home/kabouik/.itch/app-25.4.0'
21:35:55.611 INFO itch@25.4.0 on electron@2.0.8 in production
21:35:55.614 DEBUG (m/reac/preb/load-preferences) imported preferences: {"downloadSelfUpdates":true,"offlineMode":false,"installLocations":{"f8589216-5929-4894-b7ad-b30701b3075b":{"path":"/run/media/kabouik/INT 4TB/JEUX STEAM/AUTRES"}},"defaultInstallLocation":"f8589216-5929-4894-b7ad-b30701b3075b","isolateApps":false,"closeToTray":true,"readyNotification":true,"showAdvanced":true,"openAtLogin":false,"openAsHidden":false,"manualGameUpdates":true,"preventDisplaySleep":true,"preferOptimizedPatches":false,"disableBrowser":false,"enableTabs":false,"sidebarWidth":240,"onlyCompatibleGames":true,"onlyOwnedGames":false,"onlyInstalledGames":false,"layout":"grid","gotMinimizeNotification":true,"lastSuccessfulSetupVersion":"25.4.0","importedOldInstallLocations":true}
21:35:55.730 DEBUG (m/reac/preb/load-preferences) imported preferences: {"downloadSelfUpdates":true,"offlineMode":false,"installLocations":{"f8589216-5929-4894-b7ad-b30701b3075b":{"path":"/run/media/kabouik/INT 4TB/JEUX STEAM/AUTRES"}},"defaultInstallLocation":"f8589216-5929-4894-b7ad-b30701b3075b","isolateApps":false,"closeToTray":true,"readyNotification":true,"showAdvanced":true,"openAtLogin":false,"openAsHidden":false,"manualGameUpdates":true,"preventDisplaySleep":true,"preferOptimizedPatches":false,"disableBrowser":false,"enableTabs":false,"sidebarWidth":240,"onlyCompatibleGames":true,"onlyOwnedGames":false,"onlyInstalledGames":false,"layout":"grid","gotMinimizeNotification":true,"lastSuccessfulSetupVersion":"25.4.0","importedOldInstallLocations":true}
21:35:55.732 DEBUG (m/net/register-itch-protocol) Registering itch: protocol for session with user agent Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) itch/25.4.0 Chrome/61.0.3163.100 Electron/2.0.8 Safari/537.36
21:35:55.775 DEBUG (m/reac/open-at-login) Updating login item settings, open=false, hidden=false
21:35:55.776 DEBUG (m/reac/open-at-login) Copying /usr/share/applications/io.itch.itch.desktop => /home/kabouik/.config/autostart/io.itch.itch.desktop
21:35:55.837 INFO (m/reac/preboot) preboot ran in 108.00ms
21:35:55.838 DEBUG (m/reac/winds) Waiting on root window to dispatch boot
21:35:56.251 INFO (🔧 setup) Setup starting...
21:35:56.252 INFO (🔧 setup) Creating broth manager
21:35:56.254 INFO (🔧 setup) Ensuring broth dependencies, for startup
21:35:56.256 DEBUG (🔧 setup) Package (itch) got version prefix
21:35:56.424 INFO (📦 butler) (15.12.0) is chosen and valid
21:35:56.424 INFO (📦 butler) Switching to (15.12.0)
21:35:56.425 DEBUG (🔧 setup) Package (butler) got version prefix
21:35:56.425 INFO (🔧 setup) Refreshing butlerd!
21:35:56.425 INFO (🔧 setup) Spinning up butlerd instance 1...
21:35:56.430 INFO (🔧 setup) ...waiting for endpoint from butlerd instance 1...
21:35:56.442 DEBUG (📦 itch-setup) spawning /home/kabouik/.config/itch/broth/itch-setup/versions/1.18.0/itch-setup :: --version
21:35:56.459 INFO (🔧 setup) ...for butlerd instance 1 got endpoint 127.0.0.1:37959
21:35:56.461 INFO (🔧 setup) Now speaking with butlerd instance 1, version v15.12.0, built on Apr 27 2019 @ 02:40:42, ref 4baabbcdc78269e996c3eaef3ec8da4c72c29aac, endpoint 127.0.0.1:37959
21:35:56.466 DEBUG (m/reac/down/driver) Downloads driver reacting to gotButlerdEndpoint, cancelling state...
21:35:56.466 DEBUG (m/reac/down/driver-persistent-state) Not cancelling, current phase is 0
21:35:56.466 DEBUG (m/reac/down/driver) Downloads driver reacting to gotButlerdEndpoint, state is cancelled...
21:35:56.466 DEBUG (m/reac/down/driver) Download driver polling...
21:35:56.467 DEBUG (mcall) 🙏 Downloads.List
21:35:56.468 INFO (📦 itch-setup) (1.18.0) is chosen and valid
21:35:56.468 INFO (📦 itch-setup) Switching to (1.18.0)
21:35:56.468 INFO (🔧 setup) Already set up 25.4.0 once, delaying components upgrade check
21:35:56.468 DEBUG (🔧 setup) Waiting for butler promise...
21:35:56.468 DEBUG (🔧 setup) Syncing install locations...
21:35:56.469 DEBUG (mcall) 🙏 Install.Locations.List
21:35:56.469 DEBUG (🔧 setup) Package (itch-setup) got version prefix
21:35:56.471 WARN (butlerd/make-client) (butlerd) no handler for notification Log (in Install.Locations.List convo)
21:35:56.472 DEBUG (🔧 setup) Dispatching setup done!
21:35:56.472 INFO (🔧 setup) Setup done
21:35:56.473 DEBUG (mcall) 🙏 Fetch.Caves
21:35:56.474 INFO (m/reac/silent-location-scan) Scanning install locations for items...
21:35:56.474 DEBUG (mcall) 🙏 Install.Locations.Scan
21:35:56.480 INFO (m/reac/silent-location-scan) > Scanning install location /run/media/kabouik/INT 4TB/JEUX STEAM/AUTRES...
21:35:56.480 WARN (m/reac/silent-location-scan) Could not process install location /run/media/kabouik/INT 4TB/JEUX STEAM/AUTRES: open /run/media/kabouik/INT 4TB/JEUX STEAM/AUTRES: no such file or directory
github.com/itchio/butler/endpoints/install.(*scanContext).DoInstallLocation
	/builds/itchio/butler/endpoints/install/locations_scan.go:261
github.com/itchio/butler/endpoints/install.(*scanContext).DoInstallLocations
	/builds/itchio/butler/endpoints/install/locations_scan.go:245
github.com/itchio/butler/endpoints/install.(*scanContext).Do
	/builds/itchio/butler/endpoints/install/locations_scan.go:136
github.com/itchio/butler/endpoints/install.InstallLocationsScan
	/builds/itchio/butler/endpoints/install/locations_scan.go:66
github.com/itchio/butler/butlerd/messages.(*InstallLocationsScanType).Register.func1
	/builds/itchio/butler/butlerd/messages/messages.go:2177
github.com/itchio/butler/butlerd.(*Router).Dispatch.func2
	/builds/itchio/butler/butlerd/router.go:300
github.com/itchio/butler/butlerd.(*Router).Dispatch
	/builds/itchio/butler/butlerd/router.go:309
github.com/itchio/butler/cmd/daemon.(*handler).Handle
	/builds/itchio/butler/cmd/daemon/daemon.go:123
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1337
21:35:56.480 INFO (m/reac/silent-location-scan) Skipping...
21:35:56.481 INFO (m/reac/silent-location-scan) > Scanning install location /home/kabouik/.config/itch/apps...
21:35:56.481 INFO (m/reac/silent-location-scan) No items found
21:35:56.481 INFO (m/reac/silent-location-scan) Scan complete.
21:35:56.487 INFO Using tray image (/home/kabouik/.itch/app-25.4.0/resources/app.asar/src/static/images/tray/itch.png)
21:35:56.476 DEBUG (rcall) 🙏 Profile.List
21:35:56.546 DEBUG (mcall) 🙏 Fetch.Caves
21:35:56.752 DEBUG (rcall) 🙏 Profile.List
21:35:56.777 DEBUG (mcall) 🙏 Fetch.Commons
21:35:56.782 WARN (butlerd/make-client) (butlerd) no handler for notification Log (in Fetch.Commons convo)
21:35:58.756 DEBUG (m/reac/locales) Downloading fresh locale file from https://locales.itch.ovh/itch/en.json
21:35:59.139 DEBUG (m/reac/locales) HTTP GET https://locales.itch.ovh/itch/en.json: 200
21:35:59.140 DEBUG (m/reac/locales) Saving fresh en locale to /home/kabouik/.config/itch/locales/en.json
21:35:59.477 INFO (m/reac/login) Attempting saved login for profile 624055
21:35:59.478 DEBUG (mcall) 🙏 Profile.UseSavedLogin
21:35:59.483 INFO (m/reac/login) > Validating credentials...
21:36:00.401 INFO (m/reac/login) > Logged in! (online)
21:36:00.401 INFO (m/reac/login) Saved login succeeded!
21:36:00.402 INFO (m/reac/login) Login succeeded, setting up session
21:36:00.404 INFO (m/reac/login) Registering itch protocol for session persist:itchio-624055
21:36:00.404 DEBUG (m/net/register-itch-protocol) Registering itch: protocol for session with user agent Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) itch/25.4.0 Chrome/61.0.3163.100 Electron/2.0.8 Safari/537.36
21:36:00.405 INFO (m/reac/login) Restoring tabs...
21:36:00.405 DEBUG (mcall) 🙏 Profile.Data.Get
21:36:00.410 INFO (m/reac/login) Dispatching login succeeded
21:36:00.411 INFO (m/reac/login) Fetching owned keys...
21:36:00.411 DEBUG (mcall) 🙏 Fetch.ProfileOwnedKeys
21:36:00.413 DEBUG (mcall) 🙏 Fetch.Commons
21:36:00.414 DEBUG (m/reac/down/driver) Download driver polling...
21:36:00.414 DEBUG (m/reac/down/driver) Not paused... current phase: IDLE
21:36:00.415 DEBUG (m/reac/down/driver-persistent-state) IDLE => STARTING
21:36:00.415 DEBUG (mcall) 🙏 Downloads.Drive
21:36:00.415 DEBUG (m/reac/down/driver-persistent-state) STARTING => RUNNING
21:36:00.415 INFO (m/reac/down/driver-persistent-state) Downloads started
21:36:00.420 INFO (m/reac/login) Fetching fresh data...
21:36:00.420 INFO (m/reac/login) Fetching page 1
21:36:00.420 WARN (butlerd/make-client) (butlerd) no handler for notification Log (in Fetch.Commons convo)
21:36:00.421 INFO (m/reac/down/driver) Now driving downloads...
21:36:00.472 DEBUG (rcall) 🙏 Fetch.Caves
21:36:00.526 DEBUG (rcall) 🙏 Fetch.ProfileOwnedKeys
21:36:00.526 DEBUG (rcall) 🙏 Fetch.Caves
21:36:00.581 DEBUG (rcall) 🙏 Fetch.Cave
21:36:00.589 INFO (m/reac/login) Fetching page 2
21:36:00.600 DEBUG (rcall) 🙏 Fetch.ProfileOwnedKeys
21:36:00.615 WARN (butlerd/make-client) (butlerd) no handler for notification Log (in Fetch.ProfileOwnedKeys convo)
21:36:00.749 INFO (m/reac/login) Waited 330.806124ms for fetch (shared with another call)
21:36:00.750 INFO (m/reac/login) Fetched owned keys in 339.00ms
21:36:00.912 DEBUG (mcall) 🙏 Fetch.Commons
21:36:00.914 WARN (butlerd/make-client) (butlerd) no handler for notification Log (in Fetch.Commons convo)
21:36:01.744 DEBUG (mcall) 🙏 Profile.Data.Put
21:36:06.603 DEBUG (m/reac/navigation) Navigating to itch://library/owned 
21:36:06.646 DEBUG (rcall) 🙏 Fetch.ProfileOwnedKeys
21:36:07.739 DEBUG (mcall) 🙏 Profile.Data.Put
21:36:08.722 DEBUG (m/reac/winds) On window.close: close to tray enabled
21:36:10.745 DEBUG (m/brot/itch-setup) spawning /home/kabouik/.config/itch/broth/itch-setup/versions/1.18.0/itch-setup :: --appname ::: itch ::: --upgrade
21:36:10.769 DEBUG (📦 self) > 2019/06/28 23:36:10 itch-setup will log to /tmp/itch-setup-log.txt
21:36:10.769 DEBUG (📦 self) > 2019/06/28 23:36:10 =========================================
21:36:10.770 DEBUG (📦 self) > 2019/06/28 23:36:10 itch-setup "v1.18.0, built on Apr 10 2019 @ 13:30:34, ref 57838bc373361f2c8938336674247f869999e862" starting up at "2019-06-28 23:36:10.76884888 +0200 CEST m=+0.005774755" with arguments:
21:36:10.770 DEBUG (📦 self) > 2019/06/28 23:36:10 "/home/kabouik/.config/itch/broth/itch-setup/versions/1.18.0/itch-setup"
21:36:10.770 DEBUG (📦 self) > 2019/06/28 23:36:10 "--appname"
21:36:10.770 DEBUG (📦 self) > 2019/06/28 23:36:10 "itch"
21:36:10.770 DEBUG (📦 self) > 2019/06/28 23:36:10 "--upgrade"
21:36:10.770 DEBUG (📦 self) > 2019/06/28 23:36:10 =========================================
21:36:10.770 DEBUG (📦 self) > 2019/06/28 23:36:10 App name specified on command-line: itch
21:36:10.770 DEBUG (📦 self) > 2019/06/28 23:36:10 Locale:  en-US
21:36:10.771 DEBUG (📦 self) > 2019/06/28 23:36:10 Initializing installer GUI...
21:36:10.771 DEBUG (📦 self) > 2019/06/28 23:36:10 Using GTK UI
21:36:10.879 DEBUG (📦 self) > 2019/06/28 23:36:10 Initializing (itch) multiverse @ (/home/kabouik/.itch)
21:36:10.880 DEBUG (📦 self) > 2019/06/28 23:36:10 (/home/kabouik/.itch)(current = "25.4.0", ready = "")
21:36:11.195 DEBUG (📦 self) > 2019/06/28 23:36:11 Installed 25.4.0
21:36:11.195 DEBUG (📦 self) > 2019/06/28 23:36:11 Latest    25.4.0
21:36:11.195 DEBUG (📦 self) > 2019/06/28 23:36:11 We're up-to-date!
21:36:11.202 DEBUG (📦 butler) Trying to satisfy semver constraint (^15.0.1)
21:36:11.202 DEBUG (📦 butler) GET (https://broth.itch.ovh/butler/linux-amd64/versions)
21:36:11.505 INFO (📦 butler) Latest is (15.12.0)
21:36:11.505 INFO (📦 butler) Already the active version, nothing to do
21:36:11.510 DEBUG (📦 butler) Present versions: 15.12.0
21:36:11.511 DEBUG (📦 itch-setup) Trying to satisfy semver constraint (^1.8.0)
21:36:11.511 DEBUG (📦 itch-setup) GET (https://broth.itch.ovh/itch-setup/linux-amd64/versions)
21:36:11.915 INFO (📦 itch-setup) Latest is (1.18.0)
21:36:11.915 INFO (📦 itch-setup) Already the active version, nothing to do
21:36:11.917 DEBUG (📦 itch-setup) Present versions: 1.18.0
21:36:25.751 INFO (m/reac/updater) Regularly scheduled check for game updates...
21:36:25.753 INFO (m/reac/updater) Scheduling next game update check for Fri Jun 28 2019 23:57:11 GMT+0200 (CEST)
21:36:25.754 DEBUG (mcall) 🙏 CheckUpdate
21:36:25.758 INFO (m/reac/updater) Looking for updates to 4 items...
21:36:25.758 INFO (m/reac/updater) ...for runtime 64-bit Linux
21:36:26.467 INFO (m/reac/updater) < Checked 4 entries in 709.495939ms
```

**Comment by tkashkin** [comment]
Still it's probably related to libraries included in AppImage.
I recommend to use distro-specific packages or build it from source yourself to avoid similar problems.

GameHub is not available in Solus repos yet, however it's now passing Solus package policies and should probably be available soon: https://dev.getsol.us/T7514, https://dev.getsol.us/D6623.

**Comment by Kabouik** [comment]
Yes, I'm eagerly looking forward to that (I'm Matf in that Solus task and I pinged the Solus team on IRC to get their answer). It's good to know that it is probably specific to the AppImage, it should be solved soon for me then. However, I suppose the AppImage will still be an interesting solution for a number of users, hopefully it can be fixed.

Thanks for your quick help @tkashkin.

**Comment by Kabouik** [comment]
Now that Gamehub is available in the official Solus repository, I can confirm the those games or programs from the list above now run correctly: Prison Architect, Overgrowth, Low Light Combat, Gone Home, itch.io. Hopefully it will confirm what is wrong with the AppImage and help fixing it.

Hob still won't launch, but I am trying to run it with Proton and may very well have set it up incorrectly ([something worng?](https://0x0.st/zLYI.png) It's off-topic so I will open a new issue later with logs, unless you see something obvious in the screenshot).

It still frustrating to switch to it, considering how many new features have been included in the latest AppImage release.

**Comment by tkashkin** [comment]
Does it work in bd9635c?

**Comment by Kabouik** [comment]
Version: 0.14.1-bd9635c-dev
Branch:  dev
Commit:  bd9635c (bd9635c0251bba14a150645940be7f56a892fc07)
Distro:  [AppImage] Solus
DE:      Budgie:GNOME

Tried the above version, and unfortunately the issue is still here.
